client/src/diff/TextDiff.js

```diff
@@ -100,9 +98,13 @@ const process_indicators = (matched_indicators, indicator_status) => {
         };
       }
       const indicator = indicator_pair[0];
-      const status = indicator.doc === "dp18" ?
-        "indicator_removed":
-          indicator.doc === "dp19" ? "indicator_added": indicator.doc;
+      const status = indicator.doc === previous_dp_key ?
+        "indicator_removed" :
+        (
+          indicator.doc === current_dp_key ? 
+            "indicator_added" : 
+            indicator.doc
+        );
```

Comment:
Hmm, this is a semi common pattern for us, but I don't love it. Better than a `let` where the assignment happens in a following set of if statements though.

I've occasionally used an immediately invoked function to get the same single expression declaration. Does anyone else agree that the following is better than the nested ternary?
```
const status = (() => {
  switch(indicator.doc){
    case previous_dp_key:
      return 'indicator_removed'
    case current_dp_key:
      return 'indicator_added'
    default:
      return indicator.doc;
  }
})();
```

Reply:
From my perspective this results from trying to do things on one line. I'm not sure a switch construct is any better than an if/else.

Reply:
The desire is for `consts` over `lets`. `consts` give (some) assurance that the value isn't modified post-declaration. Doing things on one line/in a single expression is just a requirement of `consts`.